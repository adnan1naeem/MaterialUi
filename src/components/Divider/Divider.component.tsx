import * as React from 'react';
import { DividerProperties } from '@blueeast/bluerain-ui-interfaces';
import Divider from '@material-ui/core/Divider';


/**
 * The universal props of Divider Component
 * @param {ReactNode} props.light If true, Divider is generated with light color.
 *  The specific libraray props of Divider Component
 * @param {ReactNode} props.component component to generate Divider with.
 * @param {ReactNode} props.inset if true, Divider is generated indented.
 */

export interface MUIDividerProperties extends DividerProperties {
  component?: string;
  inset?: boolean;
  absolute?: boolean;
  classes?: {};
  light?: boolean
}
const BlueRainDivider: React.StatelessComponent<MUIDividerProperties> = (props) => {
  return (
    <Divider {...props} />
  );
};

export default BlueRainDivider;
