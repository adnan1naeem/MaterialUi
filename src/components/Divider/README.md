# Divider

A divider is a thin line that groups content in lists and layouts.

## Compatibility

| 🌏 Web | 🖥 Electron | 📱 React Native |
| :----: | :---------: | :-------------: |
| ✔️     | ✖           | ✖             |

## Screenshots

| 🌏 Web | 🖥 Electron | 📱 React Native |
| :---: | :--------: | :------------: |
| ![web image](./screenshots/Divider.png)  |    TBD   |  TBD |

## Universal Props

| Name | Type | Default | Description |
|:-----|:-----|:--------|:------------|
| light | bool | | If true, Divider is generated with light color. |

## Specific Library Props

| Name | Type | Default | Description |
|:-----|:-----|:--------|:------------|
| component | Node | hr | component to generate Divider with. |
| inset | bool |  | if true, Divider is generated indented. |
| absolute | bool | false | Position of component. |

## How to use

```JavaScript
import React from 'react';
import BR from '@blueeast/bluerain-os';

const DividerSample = (props) => (

    const Divider = BR.Components.get('Divider');
    return <Divider light={false} />;

export default DividerSample;
```