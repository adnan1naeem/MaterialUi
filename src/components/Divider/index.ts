// export { default as Divider, MUIDividerProperties } from './Divider.component';
