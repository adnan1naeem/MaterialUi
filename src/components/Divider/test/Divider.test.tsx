import 'jest';
import Divider from '../Divider.component';
import React from 'react';
import { shallow, configure } from 'enzyme'; import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('Divider component with light value props', () => {
  const divider = shallow(
    <Divider light={true} />
  );
  expect(divider.props().light).toEqual(true);
});

test('Divider component with absolute  prop=true', () => {
  const divider = shallow(
    <Divider absolute={true} />
  );
  expect(divider.props().absolute).toEqual(true);
});


test('Divider component with inset prop=true', () => {
  const divider = shallow(
    <Divider inset={true} />
  );
  expect(divider.props().inset).toEqual(true);
});

test('Divider component with inset prop=false', () => {
  const divider = shallow(
    <Divider inset={false} />
  );
  expect(divider.props().inset).toEqual(false);
});



test('Divider component with inset prop=false', () => {
  const divider = shallow(
    <Divider classes={{}} />
  );
  expect(divider.props().classes).toEqual({});
});


