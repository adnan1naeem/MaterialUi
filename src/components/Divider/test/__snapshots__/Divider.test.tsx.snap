// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component with default values 1`] = `
<hr
  className="MuiDivider-root-1"
/>
`;

exports[`Divider component with style 1`] = `
<hr
  className="MuiDivider-root-1 MuiDivider-absolute-2"
/>
`;
